### ISSUE TITLE
REST Request with user:password

### ISSUE THREAD

**Issue by rmorillo24** [taverntesting/tavern#59]
I'm trying to use tavern for testing an API based on __flask_restful__ and __flask_httpauth__. This modules use the auth header, and I think there is no support for it yet. I've done several tests and that's what i've figured out (sorry, I can't follow down your code... yet)

My server is capable of answering correctly the following CURL query:

```
curl -H "Content-type:application/json" \
    -u 'user:password' \
    -d '{"param":"value"}' \
    -X GET 'http://127.0.0.1:5000/api/v1.0/user'

```

Please notice that the login is redundant for the sake of this example
However, i've tried to include this information in the yaml file, within the _request_ and _headers_ block, but I haven't been able to receive any information on my server.
Many thanks in advance!

**Comment by Birne94** [comment]
HTTP basic authentication is done through the `Authorization` header. Curl sets this header for you which you can see if you add the `-v` (verbose) flag.

`> Authorization: Basic dXNlcjpwYXNzd29yZA==`

While not really _nice_, you can pass username and password by constructing this header manually. Wikipedia, for example, [lists the basic encoding steps](https://en.wikipedia.org/wiki/Basic_access_authentication#Client_side). Basically it is just a base64 encoded version of the string `user:password`.

**Comment by michaelboulton** [comment]
I thought the 'auth' keyword was correctly implemented for this, but it seems like it was only half-done - I've fixed the implementation in version 0.6.1 and updated the documentation here: https://taverntesting.github.io/documentation#authorisation

**Comment by rmorillo24** [comment]
That's working for me. Many thanks :)
I close the issue, although I haven't been able to test the other options you've added to the auth, besides the basic one I reported. If I have a chance to test some more and find anything I will let you know.
Cheers.
